### ISSUE TITLE
Mane or service not know

### ISSUE THREAD

**Issue by 330Speedy** [QIDITECH/QIDI_Q1_Pro#51]
![IMG_3024](https://github.com/QIDITECH/QIDI_Q1_Pro/assets/161778185/744753e7-0cae-4960-93b2-fb556d545e9d)

**Comment by konstantin-kelemen** [comment]
I did have the same issue, running `sudo dos2unix /etc/resolv.conf` fixed it.

**Comment by 330Speedy** [comment]
Great work!!! It works... Thank you very much :)
Mainsail is installed and up to date. But I can't open the web UI. The classic port 4408 doesn't work... Or does this install set another specific port?

**Comment by pbyrne234** [comment]
Hi; any update on specific ports?

**Comment by CChen616** [comment]
Checking which port is being listened to in `/etc/nginx/sites-available/mainsail` might be helpful. If it’s also listening on port 80, it may conflict with Fluidd.

**Comment by 330Speedy** [comment]
> `/etc/nginx/sites-available/mainsail`Může být užitečné zkontrolovat, na kterém portu se naslouchá . Pokud také naslouchá na portu 80, může to být v konfliktu s Fluidd.

Yes, I just checked and both mainsail and fluidd are on the same port 80. Please, how can I change the port? it didn't ask me during installation
Thank you

**Comment by CChen616** [comment]
You only need to change the port used by one of the configurations to another, such as 4180 (you can use `netstat -ntlp` to check port usage and avoid conflicts), then restart the printer.

When accessing, use
```sh
# port can be omitted if service listening on port 80, which is http default port
http://{your_printer_ip}:{port}
# for example
http://192.168.0.2:4180
```

**Comment by pbyrne234** [comment]
Hi can't find where KIAUH is installed? help appreciated.

**Comment by pbyrne234** [comment]
@CChen616 Thanks, had already found it.  Fluidd needs to be updated as it never shows correct estimated finish time always only shows start time instead.

![fluidd](https://github.com/QIDITECH/QIDI_Q1_Pro/assets/64602017/118dd92b-f15a-4ca2-9346-06e254f94110)

**Comment by CChen616** [comment]
![image](https://github.com/QIDITECH/QIDI_Q1_Pro/assets/88018764/a44b2cfa-0944-4501-8ee2-ecc05c71c0fe)

These Z-axis related buttons have been adjusted in the stock Fluidd interface. After your upgrade, their directions will be reversed:
The up arrow will make the platform move down.

**Comment by 330Speedy** [comment]
Hi everyone…. I have a little question 😉 
Is it possible to update klipper, moonraker, fluidd to the latest version? or if not, is there any plan to ship the latest version with the next FW update, or even native support for Mainsail? 
Thank you All

**Comment by pbyrne234** [comment]
@330Speedy 
Mainsail is working fine when installed with KIAUH  when you change the port in sites-available.

😉

**Comment by 330Speedy** [comment]
> @330Speedy Mainsail is working fine when installed with KIAUH when you change the port in sites-available.
> 
> 😉

Yes, Mainsail works, but because of the old version of Moonraker and Klipper, not all options in Mainsail work...

**Comment by pbyrne234** [comment]
@330Speedy Agreed they need to update both in next firmware.

**Comment by CChen616** [comment]
The machine uses custom versions of Moonraker and Klipper. Upgrading will overwrite the modified files, causing the printer to malfunction. Please avoid upgrading these components.
